Libraries/acdriver/Sources/Compile/ImageSet.cpp

```diff
@@ -151,11 +151,29 @@ CompileAsset(
 
         format = car::Rendition::Data::Format::JPEG;
     } else {
-        result->normal(
-            Result::Severity::Error,
-            "unknown file type",
-            filename);
-        return false;
+        ext::optional<Extensions::ImageType> type = Extensions::ImageTypeFromFileExtension(FSUtil::GetFileExtension(filename));
+        if (!type) {
+            result->normal(
+                Result::Severity::Error,
+                "unknown file type",
+                filename);
+            return false;
+        }
+        if (!compileOutput->allowedExtensionsImageTypes().count(*type)) {
+            result->normal(
+                Result::Severity::Error,
+                "forbidden file type",
+                filename);
+            return false;
+        }
+        if (!filesystem->read(&pixels, filename)) {
+            result->normal(
+                Result::Severity::Error,
+                "unable to read image file",
+                filename);
+            return false;
+        }
+        format = Extensions::ImageTypeToDataFormat(*type);
```

Comment:
Why are extension formats any different from JPEG and PNG (lines 100-152)? Why not write the WebP-handling code like this to match how JPEG files are handled:

```cpp
     } else if (FSUtil::IsFileExtension(filename, "webp", true)) {
         if (!compileOutput->isWebpAllowed()) {
             result->normal(
                 Result::Severity::Error,
                 "forbidden file type",
                 filename);
             return false;
         }
         if (!filesystem->read(&pixels, filename)) {
             result->normal(
                 Result::Severity::Error,
                 "unable to read WebP file",
                 filename);
             return false;
         }
 
         format = car::Rendition::Data::Format::WEBP;
    } else {
 ``

Reply:
PNG is handled different from everything else, but the difference with jpeg is just that's one if standard, the other one is not...